src/main/scala/cromwell/engine/backend/jes/authentication/JesAuthentication.scala

```diff
@@ -0,0 +1,66 @@
+package cromwell.engine.backend.jes.authentication
+
+import cromwell.engine.WorkflowDescriptor
+import cromwell.engine.backend.jes.{JesConfigurationSingleton, JesConfiguration, JesInterface}
+import spray.json.JsObject
+
+import scala.util.{Failure, Success}
+
+/**
+ * This makes JesConnection non visible when mixing a JesAuthentication trait
+ */
+trait JesConnection {
+  val JesConnection: JesInterface
+}
+
+/**
+ * Trait for JesAuthentication
+ */
+trait JesAuthentication { self: JesConfiguration with JesConnection =>
+
+  private def getRefreshToken(descriptor: WorkflowDescriptor) = descriptor.workflowOptions.get("refresh_token")
+  private def getAccessToken(descriptor: WorkflowDescriptor) = descriptor.workflowOptions.get("access_token")
+
+  def authenticated[A](f: JesInterface => A)(implicit descriptor: WorkflowDescriptor) = {
+    val interface = JesConf.authMode match {
+      case RefreshTokenMode => getRefreshToken(descriptor) match {
+        case Success(token) => JesConnection.withRefreshToken(token)
+        case Failure(error) => throw new RuntimeException(s"Error retrieving refresh token : ${error.getMessage}")
+      }
+      case AccessTokenMode => getAccessToken(descriptor) match {
+        case Success(token) => JesConnection.withAccessToken(token)
+        case Failure(error) => throw new RuntimeException(s"Error retrieving access token : ${error.getMessage}")
+      }
+      case _ => JesConnection
```

Comment:
We should only leave the stuff we need to actually support all of this. we can always dig this back up if necessary